gradle-diff-task-dependencies/plugin/src/main/groovy/com/novoda/gradle/diffdependencytask/ConditionalDependency.groovy

```diff
@@ -0,0 +1,33 @@
+package com.novoda.gradle.diffdependencytask
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+import java.util.regex.Pattern
+
+public class ConditionalDependency {
+
+    static {
+        Project.metaClass.registerConditionalDependency << { ConditionalDependency conditionalDependency ->
```

Comment:
Only this class uses this custom method, so adding it here will make it possible to test this class without running custom registrations.